llrt_core/src/modules/crypto/mod.rs

```diff
@@ -132,6 +133,49 @@ fn random_fill_sync<'js>(
     Ok(obj)
 }
 
+macro_rules! fill_typed_array {
+    ($ty:ty, $bytes:expr, $rng:expr) => {{
+        let size = mem::size_of::<$ty>();
+        for chunk in $bytes.chunks_exact_mut(size) {
```

Comment:
Why do we need to chunk? Can't we fill an entire buffer?

Reply:
Are you suggesting that we fill the entire length of the array with random numbers, one byte at a time?

Each array buffer type has a primitive type size for each element, and I thought that if I filled a completely random number, it might end up with a value that could not exist in that type.

I tried that approach in this effort (it may have been an issue with my implementation...), it sometimes resulted in out-of-range asserts in the tests.

Reply:
I think we can use something like this no?

https://rust-random.github.io/rand/rand/trait.RngCore.html#tymethod.fill_bytes

Reply:
And since no true random is needed we can use the global inited SYSTEM_RANDOM variable

Reply:
> And since no true random is needed we can use the global inited SYSTEM_RANDOM variable

I do not understand how using SYSTEM_RANDOM would solve this problem...

Reply:
It won’t solve this problem just another perf suggestion. We can skip Float Typed Arrays and fill buffers using SYSTEM_RANDOM

Reply:
Hi @richarddavison , I have responded, please review.